### ISSUE TITLE
False positive "Uncallable method" when declaring record within a method

### ISSUE THREAD

**Issue by baardsen** [spotbugs/spotbugs#1569]
When I declare a `record` within another method and only reference the generated accessor methods using method references then I get a `UMAC_UNCALLABLE_METHOD_OF_ANONYMOUS_CLASS` failure even though the method is already being used.

```java
import java.util.function.Supplier;

public class UncallableMethodOfRecord {

    public static void main(String[] args) {
        record Person(String name) { }
        Person alice = new Person("Alice");
        print(alice::name);
    }

    private static void print(Supplier<String> supplier) {
        System.out.println("Hello " + supplier.get());
    }

}
```

Output:
```
H C UMAC: Uncallable method UncallableMethodOfRecord$1Person.name() defined in anonymous class  At UncallableMethodOfRecord.java:[line 6]
```

**Comment by gtoison** [comment]
This was fixed by #2170
